lms/djangoapps/support/views/course_entitlements.py

```diff
@@ -0,0 +1,144 @@
+            CourseEntitlementSupportDetail.objects.create(
+                entitlement=entitlement, reason=reason, comments=comments, unenrolled_run=unenrolled_run,
+                support_user=support_user
+            )
+            return Response(
+                status=status.HTTP_201_CREATED,
+                data=SupportCourseEntitlementSerializer(instance=entitlement).data
+            )
+        else:
+            return HttpResponseBadRequest(
+                u'Could not find an entitlement for user {username} in course {course}'.format(
+                    username=username_or_email,
+                    course=course_uuid
+                )
+            )
+
+    @method_decorator(require_support_permission)
+    def create(self, request, *args, **kwargs):
+        """ Allows support staff to grant a user a new entitlement for a course. """
+        support_user = request.user
+        try:
+            username_or_email = self.request.data['user']
+            user = User.objects.get(Q(username=username_or_email) | Q(email=username_or_email))
+            course_uuid = request.data['course_uuid']
+            reason = request.data['reason']
+            comments = request.data.get('comments', None)
+            mode = request.data.get('mode', 'verified')
+        except KeyError as err:
+            return HttpResponseBadRequest(u'The field {} is required.'.format(text_type(err)))
+        except User.DoesNotExist:
+            return HttpResponseBadRequest(
+                u'Could not find user {username}.'.format(
+                    username=username_or_email,
+                )
+            )
+        if CourseEntitlement.get_entitlement_if_active(user, course_uuid) is not None:
+            return HttpResponseBadRequest(u'User {user} has an active entitlement in course {course}.'.format(
+                user=username_or_email, course=course_uuid
+            ))
+        else:
+            entitlement = CourseEntitlement.objects.create(user=user, course_uuid=course_uuid, mode=mode)
+            CourseEntitlementSupportDetail.objects.create(
+                entitlement=entitlement, reason=reason, comments=comments, support_user=support_user
+            )
+            return Response(
+                status=status.HTTP_201_CREATED,
+                data=SupportCourseEntitlementSerializer(instance=entitlement).data
+            )
+
+    @staticmethod
+    def get_most_recent_entitlement(user, course_uuid):
+        """
+        Returns the most recently created entitlement for the given user in the given course.
+
+        Args:
+            user (User): user object record for which we are retrieving the entitlement.
+            course_uuid (UUID): identified of the course for which we are retrieving the learner's entitlement.
+        """
+        return CourseEntitlement.objects.filter(user=user, course_uuid=course_uuid).latest('created')
+
+    @staticmethod
```

Comment:
This method should probably be in the CourseEntitlement class within the models.py file.